pkgs/by-name/le/lean-lsp-mcp/package.nix

```
@@ -0,0 +1,90 @@
+{
+  lib,
+  python3,
+  fetchFromGitHub,
+}:
+
+let
+  leanclient = python3.pkgs.buildPythonPackage rec {
+    pname = "leanclient";
+    version = "0.1.12";
+    pyproject = true;
+
+    src = python3.pkgs.fetchPypi {
+      inherit pname version;
+      sha256 = "sha256-yPALTn0WG9D6fT6jDInRt0y10/iyNIUuPEo42srLcIY=";
+    };
+
+    build-system = with python3.pkgs; [
+      poetry-core
+    ];
+
+    dependencies = with python3.pkgs; [
+      orjson
+      tqdm
+    ];
+
+    doCheck = false;
+
+    meta = {
+      description = "Lean client library for Python";
+      homepage = "https://pypi.org/project/leanclient/";
+      license = lib.licenses.mit;
+    };
+  };
+
+  mcp = python3.pkgs.mcp.overridePythonAttrs (old: rec {
+    version = "1.10.0";
+    src = python3.pkgs.fetchPypi {
+      pname = "mcp";
+      inherit version;
+      sha256 = "sha256-kfsWI8P68UV3Yj0UdV0yE9uDfF2l2uhQaeG1kSTL4Ok=";
+    };
+    dependencies =
+      (old.dependencies or [ ])
+      ++ (with python3.pkgs; [
+        jsonschema
+        python-multipart
+      ]);
+    dontCheckRuntimeDeps = true;
+    doCheck = false;
+  });
+
+in
+python3.pkgs.buildPythonApplication rec {
+  pname = "lean-lsp-mcp";
+  version = "0.3.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "oOo0oOo";
+    repo = pname;
```

```suggestion
    repo = "lean-lsp-mcp";
```
Do not rely on pname and repo name matching